assets/js/modules/analytics/datastore/properties.test.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * modules/analytics data store: properties tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Node dependencies
+ */
+import fs from 'fs';
+import path from 'path';
+
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { STORE_NAME } from './index';
+import {
+	createTestRegistry,
+	muteConsole,
+	subscribeUntil,
+	unsubscribeFromAll,
+} from 'tests/js/utils';
+
+describe( 'modules/analytics properties', () => {
+	const fixtures = {
+		// TODO: Use consolidated fixture data.
+		accountsPropertiesProfiles: JSON.parse( fs.readFileSync( path.join( __dirname, '__fixtures__', 'accounts-properties-profiles.json' ) ) ),
+		createProperty: JSON.parse( fs.readFileSync( path.join( __dirname, '__fixtures__', 'create-property.json' ) ) ),
+		propertiesProfiles: JSON.parse( fs.readFileSync( path.join( __dirname, '__fixtures__', 'properties-profiles.json' ) ) ),
+		profiles: JSON.parse( fs.readFileSync( path.join( __dirname, '__fixtures__', 'profiles.json' ) ) ),
+	};
+	let apiFetchSpy;
+	let registry;
+	let store;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		store = registry.stores[ STORE_NAME ].store;
+
+		apiFetchSpy = jest.spyOn( { apiFetch }, 'apiFetch' );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+		apiFetchSpy.mockRestore();
+	} );
+
+	describe( 'actions', () => {
+		describe( 'createProperty', () => {
+			it( 'creates a property and adds it to the store ', async () => {
+				const accountId = fixtures.accountsPropertiesProfiles.accounts[ 0 ].id;
+
+				fetch
+					.doMockIf(
+						/^\/google-site-kit\/v1\/modules\/analytics\/data\/create-property/
+					)
+					.mockResponse(
+						JSON.stringify( fixtures.createProperty ),
+						{ status: 200 }
+					);
+
+				registry.dispatch( STORE_NAME ).createProperty( accountId );
+				// TODO: This is failing and I'm not clear on why, when it works fine
+				// in `assets/js/googlesitekit/datastore/site/reset.test.js`
+				// expect( registry.select( STORE_NAME ).isDoingCreateProperty( accountId ) ).toEqual( true );
```

Comment:
🚨🚨🚨

This isn't working as expected and I'm not clear on why. I'm guessing I'm just missing something obvious from starring at this branch for too long, so other 👀 appreciated! 🙂

Reply:
It's because you're dispatching `createProperty`. By the time the assertion runs, it's already done so it is correct.

In `reset.test.js` you dispatched the `fetch*` action directly.
https://github.com/google/site-kit-wp/blob/8186dbfd03b8a5339cff8356f12eed7ee69bd588/assets/js/googlesitekit/datastore/site/reset.test.js#L69-L70

If I change the above to `fetchCreateProperty` and bypass the rest of the test, it passes 😄 

I'll add a new test for this separately as in the reset test 👍